src/wasm/wat-parser.cpp

```diff
@@ -1450,6 +1466,22 @@ struct ParseDefsCtx : TypeParserCtx<ParseDefsCtx> {
     return name;
   }
 
+  Result<uint32_t> getDataFromIdx(uint32_t idx) {
+    if (idx >= wasm.dataSegments.size()) {
+      return in.err("data index out of bounds");
+    }
+    return idx;
+  }
+
+  Result<uint32_t> getDataFromName(Name name) {
```

Comment:
Could this perhaps be `getDataIndexFromName` and the above be `getValidDataIndex`?

Reply:
These should be consistent with the other `getXFromIdx` and `getXFromName` functions, so if we rename these we would want to rename them all. I would rather do that in a separate PR if we want to that at all.

I'm guessing the motivation here is that fact that data segments are referred to by index rather than by Name in Binaryen IR, but that's an unnecessary inconsistency that I would like to fix at some point.

Reply:
Followup sounds good.

(To me, reading `getData` was not clear enough to understand what this was actually getting, which is the index.)